### PR TITLE
Add a varibale of library mode when installing plugin

### PR DESCRIPTION
**Pull request by fujunwei**
Use the parameter of LIB_MODE to specify the usage of Crosswalk runtime library
when installing the plugin, Use shared mode like --variable LIB_MODE="shared".

Also user can specify different Crosswalk version using a <preference> tag within your config.xml
The version of 13 in embedd mode
<preference name="xwalkVersion" value="org.xwalk:xwalk_core_library_beta:13+" />
<preference name="xwalkVersion" value="xwalk_core_library_beta:13+" />
The version of 13 in shared mode
<preference name="xwalkVersion" value="org.xwalk:xwalk_shared_library_beta:13+" />
<preference name="xwalkVersion" value="xwalk_shared_library_beta:13+" />
The used mode will depend on the preference of {{lib_mode}}
<preference name="xwalkVersion" value="13+" />
<preference name="xwalkVersion" value="13" />

### CODE REVIEW

**Comment by fujunwei** [comment]
If we use embedd mode, we must append following command when installing plugin
--variable LIB_MODE="embedd"

**Comment by fujunwei** [comment]
@huningxin PTAL, thanks.

**Comment by huningxin** [comment]
I prefer to avoid changing the default behavior which uses "embedded" mode. Probably, we can allow developer to configure this in config.xml. Any thoughts?

**Comment by baleboy** [comment]
agree about not changing the default behavior

**Comment by fujunwei** [comment]
Thanks, I will configure this in config.xml.

**Comment by fujunwei** [comment]
@huningxin PTAL.

File: plugin.xml [hunk 1]
```diff
@@ -19,6 +19,11 @@
     <platform name="android">
         <config-file target="res/xml/config.xml" parent="/*">
             <preference name="webView" value="org.crosswalk.engine.XWalkWebViewEngine" />
+            <preference name="lib_mode" value="embedded" />
+        </config-file>
+
+        <config-file target="AndroidManifest.xml" parent="/manifest">
+            <uses-permission android:name="android.permission.WRITE_EXTERNAL_STORAGE"></uses-permission>
```
Comment (huningxin) [inline plugin.xml]: I think the default embedded mode doesn't need the `android.permission.WRITE_EXTERNAL_STORAGE` permission.

**Comment by fujunwei** [comment]
@huningxin @lincsoon PTAL.

**Comment by huningxin** [comment]
LGTM. Thanks!

**Pull request merged**